### PR TITLE
Draft of section on higher-topos interpretation

### PR DESCRIPTION
**Pull request by mikeshulman**
Let me know what you think.  As is usually the case, it turned out rather longer than I expected.  Probably much too long.  But it's always easier to write more and then trim afterwards.

### CODE REVIEW

File: ZTors.tex [hunk 1]
```diff
@@ -1102,6 +1102,78 @@ \subsection{Induction in $\TorZ$}\label{sec:TorZ-induction}
 \cp{\gamma}(\rrfl\cto(p\cto\rrfl))&= p. 
 \end{align*}
 
+\section{Interpretation in higher toposes}
+\label{sec:topos}
+
+Voevodsky's pioneering work~\cite{1211.2851} constructed interpretations of the rules of univalent foundations (but not the entire formal system~\cite{voevodsky:not-interp}) in the Quillen model category of simplicial sets, which is a presentation of the fundamental $(\infty,1)$-topos of $\infty$-groupoids.
+After a decade of further work, this interpretation has now been extended to include a class of model categories presenting all $(\infty,1)$-topos by Shulman~\cite{shulman:univinj}, and made into an interpretation of the entire formal system by Brunerie, de Boer, Lumsdaine, and M\"{o}rtberg~\cite{initiality}.
```
Comment (DanGrayson) [inline ZTors.tex:1109]: Re: all $(\infty,1)$-topos 

Should be plural.

File: ZTors.tex [hunk 2]
```diff
@@ -1102,6 +1102,78 @@ \subsection{Induction in $\TorZ$}\label{sec:TorZ-induction}
...
+Therefore, since it classifies $\ZZ$-torsors, so does any other such homotopy colimit.
+
+Most of the work needed for this approach inside of type theory has already been done by Sojakova~\cite{sojakova:hits-hias}.
+She defines an ``$\dS$-algebra'' to be a type $C$ equipped with a point $c:C$ and a path $s:c=_C c$, and defines an $\dS$-algebra to ``have $\dS$-induction'' precisely if it satisfies the analogue of our~\eqref{eq:circle-induction} (see~\cite[Notation 22]{sojakova:hits-hias}).
+Her main theorem~\cite[Theorem 50]{sojakova:hits-hias} then implies that if $C$ has $\dS$-induction, then it is \emph{homotopy-initial} in that for any other $\dS$-algebra $(A,a,l)$ the type of ``$\dS$-homomorphisms''
+\[H_{C,A} \defeq \sum_{h:C\to A} \sum_{\beta: h(c) = a} \ap{h}(s) \ct \beta = \beta \ct l  \]
+is contractible, i.e.\ there is an element of the type
```
Comment (DanGrayson) [inline ZTors.tex:1127]: Commas are needed after every "i.e.".

File: ZTors.tex [hunk 3]
```diff
@@ -1102,6 +1102,78 @@ \subsection{Induction in $\TorZ$}\label{sec:TorZ-induction}
...
+Since path objects supply the interpretation of identity types, this corresponds to a type $A$ with an element $a:A$ and path $l:a=_A a$, i.e.\ an $\dS$-algebra.
+By a similar argument, the homotopy fiber that we want to prove contractible is equivalent to the hom-space $\sE(1, H_{\TorZ,A})$.
+Thus, it suffices to observe that if $B$ is an object with a point $1\to \mathrm{Contr}(B)$ then $B$ is equivalent to the terminal object (a detailed proof can be found in~\cite[Lemma 4.1]{shulman:elreedy}).
+
+It is also worth saying a few words about why our $\TorZ$ ``classifies $\ZZ$-torsors'' in the $(\infty,1)$-categorical sense, i.e.\ that for any object $A$, the hom-space $\sE(A,\TorZ)$ is naturally equivalent to the $\infty$-groupoid of small $\ZZ$-torsors in $\sE/A$.
+Importantly, the relevant notion of ``$\ZZ$-torsor'' is the topos- and sheaf-theoretic one: a $\ZZ$-torsor $X$ in $\sE/A$ need not be ``globally'' isomorphic to $\ZZ\times A$, but only ``locally'' so in the sense that there is an effective epimorphism $p:B \twoheadrightarrow A$ such that $p^*X$ is isomorphic over $B$ to $\ZZ\times B$.
```
Comment (DanGrayson) [inline ZTors.tex:1144]: "topos-" isn't a word.  What about "topos-theoretic" instead?

File: ZTors.tex [hunk 4]
```diff
@@ -1102,6 +1102,78 @@ \subsection{Induction in $\TorZ$}\label{sec:TorZ-induction}
...
+
+To start with, we observe that by the univalence axiom, $\UU$ is an object classifier, in the sense that $\sE(A,\UU)$ is naturally equivalent, by pullback of the canonical map $\UUp\to \UU$, to a full sub-$\infty$-groupoid of the core of $\sE/A$ (the fiberwise ``small'' objects).
+This means that for any $f,g:A\to \UU$, the induced map from the space of homotopies $f\sim g$ to the space of equivalences $f^*\UUp \equiv g^*\UUp$ in $\sE/A$ is an equivalence.
+This holds because the former is the space of lifts of $(f,g):A\to \UU\times \UU$ to the path-object $P\UU$, while the latter is (e.g.\ by~\cite[Lemma 4.3]{shulman:elreedy}) the space of sections of $\mathrm{Equiv}_A(f^*\UUp,g^*\UUp)$, or equivalently the lifts of $(f,g)$ to $\mathrm{Equiv}_{\UU\times \UU}(\pi_1^*\UUp,\pi_2^*\UUp)$, and the univalence axiom says precisely that $P\UU$ is equivalent to $\mathrm{Equiv}_{\UU\times \UU}(\pi_1^*\UUp,\pi_2^*\UUp)$ over $\UU\times\UU$.
```
Comment (DanGrayson) [inline ZTors.tex:1148]: Every "e.g." needs a comma after it.

File: ZTors.tex [hunk 5]
```diff
@@ -1102,6 +1102,78 @@ \subsection{Induction in $\TorZ$}\label{sec:TorZ-induction}
...
+
+Similarly, $\sum_{X:\UU} (X\to X)$ is the exponential in $\sE/\UU$ of $\UU_p$ by itself, so by the pullback-stability and universal property of exponentials, the space of lifts of $f:A\to \UU$ to $\sum_{X:\UU} (X\to X)$ is equivalent to the space of endomorphisms of the corresponding object $f^*\UUp$ of $\sE/A$.
+Thus, $\sum_{X:\UU} (X\to X)$ classifies objects equipped with an endomorphism, in that $\sE(A, \sum_{X:\UU} (X\to X))$ is naturally equivalent to the $\infty$-groupoid of small objects of $\sE/A$ equipped with an endomorphism.
+
+Finally, we need to consider the topos-theoretic interpretation of propositional truncation.
+We can ignore its particular construction in {\UniMath} and focus on its universal property, which says that it is a reflection into propositions, i.e.\ for any proposition $P$ we have $(\Trunc{X}\to P) \equiv (X\to P)$.
+Interpreted fiberwise in an $(\infty,1)$-topos, this says that if $X\to A$ is any map then $\Trunc{X}\to A$ is a reflection into the sub-$(\infty,1)$-category of $\sE/A$ consisting of the monomorphisms, i.e.\ maps $P\to A$ whose diagonal $P\to P\times_A P$ is an equivalence.
```
Comment (DanGrayson) [inline ZTors.tex:1155]: Re: "$\Trunc{X}\to A$ is a reflection"

I think you mean that the functor that sends $X\to A$ to $\Trunc{X}\to A$ is a reflection, etc.

File: ZTors.tex (hunk 2 above)
Comment (mikeshulman) [reply ZTors.tex:1127]: Is that your convention?  I didn't see any other "i.e."s in the paper, so I went with my convention.  (Although now I see that there's one "e.g." that is followed by a comma, so I could have guessed from that.)

File: ZTors.tex (hunk 3 above)
Comment (mikeshulman) [reply ZTors.tex:1144]: It's a [suspended hyphen](https://en.wikipedia.org/wiki/Hyphen#Suspended_hyphens).

File: ZTors.tex (hunk 5 above)
Comment (mikeshulman) [reply ZTors.tex:1155]: How about "$\Trunc{X}\to A$ is _its_ reflection"?

**Comment by UlrikBuchholtz** [comment]
This is great material, thanks! However, it's a bit long as it is without any subsectioning or theorem-proof waypoints. I'd suggest marking the argument starting with “Now suppose we have a Quillen…” as a theorem: TZ has the universal property of the circle. (Or corollary?)

Likewise, I'd prefer to have the material starting with “It is also worth…” as a theorem: TZ classifies Z-torsors in the sense…

But putting these in separate subsections would also work to break things up.

File: ZTors.tex (hunk 2 above)
Comment (DanGrayson) [reply ZTors.tex:1127]: I like it, because then you don't have to futz with "\ ", and line wrapping can happen.  Also, it seems to make sense in English, as well, if you think of the phrase inserted as being " ..., or, in other words, ... ".

File: ZTors.tex (hunk 3 above)
Comment (DanGrayson) [reply ZTors.tex:1144]: Okay.

File: ZTors.tex (hunk 5 above)
Comment (DanGrayson) [reply ZTors.tex:1155]: That makes it sound like reflections are always unique.

**Comment by DanGrayson** [comment]
It may be long, but it's interesting, so let's not shorten it.

File: ZTors.tex (hunk 5 above)
Comment (mikeshulman) [reply ZTors.tex:1155]: They are!

Comment (DanGrayson) [reply ZTors.tex:1155]: Well, then, that's fine.

Comment (DanGrayson) [reply ZTors.tex:1155]: Ah, somehow I was thinking of "retraction" instead.

**Comment by marcbezem** [comment]
Thanks, Mike, great contribution! Please make yourself co-author.
Here come my comments, all minor.

- p. 18, l. 7: "this paper" ---> "our paper" (to avoid confusion with the last paper you cite)

Actually, "the result of our paper" is a still ambiguous: TZ contains propositional truncation, for which we have the HoTT book version and the UniMath version. Mike asked a question about this that I feel we did not properly answer. In the paper, at least I had the HoTT book version in mind, and AFAIK we don't use propositional resizing. The HoTT book version of propositional truncation is a parametrized HIT, and footnote 10 warns against that. On the other hand, the formalization uses the UniMath version. I checked all files in UniMath/SyntheticHomotopyTheory and there is no mention of the use of propositional resizing. However, the proofs make extensive use of other results, and I'm not sure that mentioning the use of RR1 and RR2 is propagated. Also, since propositional resizing is implemented by leaving out universe consistency checks, it is completely silent (not even an interface). So this still feels uncomfortable.

- p. 19, in last line of first para, a comma before "is contractible" would not be wasted.

- p. 20, l. 7: there is an unexpected subscript "p" in "U_p", could it be "*" ?

The proof could have some more structure, the para "Finally," is not the last. There could be a sentence in the beginning of the proof making the structure explicit.

Actually, I'm puzzled by the para starting with "Finally," in relation to footnote 10, since propositional truncation IS a parametrized HIT.
Instead of "first projection from a dependent type" I would prefer "sum" over "type". The "which" in "which we may write" does not refer to the subject of the preceding subclause. Perhaps mention sum of fibers? Anyway, the abbreviation is not used very often in the sequel.

I have to look up "effective" to understand what that means in this context.

That's all.

**Comment by mikeshulman** [comment]
Thanks for the suggestions!  I've implemented most of them, and added
myself.

I don't think an extra comma on p19 would be grammatical.  I agree the
sentence is long and kind of hard to parse, though; maybe we can find
another way to re-word it.

Regarding propositional truncation, does the paper proof use the
computation rule anywhere?  (Evidently the formalization does not, since it
doesn't hold with the UniMath definition of propositional truncation.)  If
not, then my suggestion would be to just assume we have some "propositional
truncation" operation, without a computation rule, and which increases the
universe level.  This could be either a HIT or the UniMath impredicative
construction.  In an (oo,1)-topos we have both constructions, and since we
have propositional resizing (up to equivalence) we can always replace the
truncation by an equivalent one in a smaller universe, at the cost of
perhaps losing the computation rule (if we had it in the first place).
(It's not completely trivial that we have propositional resizing in a
topos, but I can add a bit about that if needed.)

(The exposition of propositional resizing is a bit disconnected currently,
by the way -- there's some discussion on p6 and then some more in section
5.  Of course it makes sense to say something introductory in the
introduction, but the formal discussion surrounding Definition 5.2 seems a
bit hidden.  Could it go in section 2?  I don't really understand what the
title of section 2 means, but it seems to be a general "introduction of
some HoTT/UniMath notions" sections.)
